File: SNAKEEAT.cpp
// O (q log n)
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
#endif

	int t;
	cin >> t;
	while (t--) {
		int n, q;
		cin >> n >> q;
		vector<long> v(n);
		for (int i = 0; i < n; i++)
			cin >> v[i];
		sort(v.begin(), v.end());
		vector<long long> sum(n);
		sum[0] = v[0];
		for (int i = 1; i < n; i++)
			sum[i] = sum[i-1] + v[i];
		while (q--) {
			long k;
			cin >> k;
			int pos = lower_bound(v.begin(), v.end(), k) - v.begin();
			int l = 0, r = pos;
			while (l < r) {
				int m = l + (r - l) / 2;
				long long need = k * (pos - m) - (sum[pos-1] - sum[m-1]);
				if (need <= m)
					r = m;
				else
					l = m + 1;
			}
			cout << n - r << endl;
		}
	}
}



// O ((n + q) * log n)
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
#endif

	int t;
	cin >> t;
	while (t--) {
		int n, q;
		cin >> n >> q;
		vector<long long> v(n+1), ans(q);
		vector<pair<long long, int>> Q(q);
		for (int i = 1; i <= n; i++)
			cin >> v[i];
		for (int i = 0; i < q; i++) {
			cin >> Q[i].first;
			Q[i].second = i;
		}
		sort(v.begin(), v.end());
		sort(Q.begin(), Q.end());
		long long prev = 0, preSum = 0;
		for (int i = 1, cur = 0; cur < q; i++, cur++) {
			long long F = Q[cur].first;
			while (v[i] < F && i <= n)
				preSum += (F - v[i++]);
			i--;
			while (preSum > prev)
				prev++, preSum -= (F - v[prev]);
			ans[Q[cur].second] = n - prev;
			if (cur < q - 1)
				preSum = preSum + (i - prev) * (Q[cur+1].first - F);
		}
		for (int i = 0; i < q; i++)
			cout << ans[i] << endl;
	}
}

File: SNAKEEAT.py
# O (q log n)
import bisect as bs

for _ in range(int(input())):
	n, q = map(int, input().split())
	ls = list(map(int, input().split()))
	ls.sort()
	Sum = [0] * n
	Sum[0] = ls[0]
	for i in range(1, n):
		Sum[i] = Sum[i-1] + ls[i]
	for __ in range(q):
		k = int(input())
		pos = bs.bisect_left(ls, k)
		l, r = 0, pos
		while l < r:
			m = l + (r - l) // 2
			need = k * (pos - m) - (Sum[pos-1] - Sum[m-1])
			if need <= m:
				r = m
			else:
				l = m + 1
		print(n - r)



# O ((n + q) log n)
for _ in range(int(input())):
	n, q = map(int, input().split())
	ls = [0] + list(map(int, input().split()))
	Q = [[int(input()), i] for i in range(q)]
	ans = [0] * q
	ls.sort()
	Q.sort()
	prev, prevSum, cur, i = 0, 0, 0, 1
	while cur < q:
		F = Q[cur][0]
		while i <= n and ls[i] < F:
			prevSum += (F - ls[i])
			i += 1
		i -= 1
		while prevSum > prev:
			prev += 1
			prevSum -= (F - ls[prev])
		ans[Q[cur][1]] = n - prev
		if cur < q - 1:
			prevSum = prevSum + (i - prev) * (Q[cur+1][0] - F)
		i += 1
		cur += 1
	print(*ans, sep = "\n")
